geowebcache/wms/src/test/java/org/geowebcache/service/wms/WMSTileFuserTest.java

```diff
@@ -120,6 +125,60 @@ public void testTileFuserResolution() throws Exception {
         assertEquals(10, tileFuser.srcIdx);
     }
 
+    @Test
+    public void testTileFuserLazyCanvasLoading() throws Exception {
+        WMSLayer layer = createWMSLayer();
+        LockProvider lockProvider = mock(LockProvider.class);
+        LockProvider.Lock lock = mock(LockProvider.Lock.class);
+        doReturn(lock).when(lockProvider).getLock(anyString());
+        layer.setLockProvider(lockProvider);
+        // request fits inside -30.0,15.0,45.0,30
+        BoundingBox bounds = new BoundingBox(-25.0, 17.0, 40.0, 22);
+        // One in between
+        int width = (int) bounds.getWidth() * 10;
+        int height = (int) bounds.getHeight() * 10;
+        GridSubset gridSubset = layer.getGridSubset(layer.getGridSubsets().iterator().next());
+        TileLayerDispatcher tld = mock(TileLayerDispatcher.class);
+        Mockito.when(tld.getTileLayer("test:layer")).thenReturn(layer);
+        StorageBroker sb = mock(StorageBroker.class);
+        WMSTileFuser tileFuser =
+                new WMSTileFuser(tld, sb, fuserRequest(layer, gridSubset, bounds, width, height));
+        AtomicInteger count = new AtomicInteger();
+        Mockito.when(sb.get(ArgumentMatchers.any(TileObject.class)))
+                .thenAnswer(
+                        invoc -> {
+                            TileObject stObj = (TileObject) invoc.getArguments()[0];
+                            final File imageTile =
+                                    new File(getClass().getResource("/image.png").toURI());
+                            stObj.setBlob(new FileResource(imageTile));
+                            stObj.setCreated((new Date()).getTime());
+                            stObj.setBlobSize(1000);
+                            if (count.incrementAndGet() == 1) {
+                                // first time, canvas should be null
+                                assertNull(tileFuser.bufferedImageWrapper.canvas);
+                                return true;
+                            } else {
+                                count.incrementAndGet();
```

Comment:
Double increment here: the first increment happened in the if, it happens whether the condition is matched, or not.